### PR TITLE
nixos/jellyseerr: init

### PR DESCRIPTION
**Pull request by camillemndn**
###### Description of changes

Adds a NixOS module to configure and run the ```jellyseerr``` service. This depends on https://github.com/NixOS/nixpkgs/pull/216037.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/misc/jellyseerr.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.services.jellyseerr;
+in
+{
+  options = {
```
Comment (JulienMalka) [inline nixos/modules/services/misc/jellyseerr.nix:8]: You can add yourself as maintainer for this module, this way you'll get pinged for future PR regarding this module.

```suggestion
  meta.maintainers = [ lib.maintainers.camillemndn ];

  options = {
```

**Review by JulienMalka — COMMENTED** [review]
It seems that jellyseerr supports a few options set via environment variables, maybe some of them could be interesting to support? (changing the port and changing media server type for example).

**Review by pennae — COMMENTED** [review]
service looks good, we still have one question though. nothing major.

File: nixos/modules/services/misc/jellyseerr.nix [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      description = "Jellyseerr, a requests manager for Jellyfin";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      environment.PORT = toString cfg.port;
+      serviceConfig = {
+        Type = "exec";
+        StateDirectory = "jellyseerr";
+        WorkingDirectory = "${pkgs.jellyseerr}/libexec/jellyseerr/deps/jellyseerr";
+        DynamicUser = true;
+        SupplementaryGroups = cfg.extraGroups;
```
[the remark this answers is not in the document]
Comment (camillemndn) [reply nixos/modules/services/misc/jellyseerr.nix]: You're right, I removed this and the related option!

**Comment by camillemndn** [comment]
@pennae It's my bad, I edited the first message so you could not see.

**Comment by camillemndn** [comment]
@pennae It is my bad, I should not have edited my first message because then you didn't see it!

**Pull request merged**